### PR TITLE
Fix errors and use promises

### PR DESCRIPTION
**Pull request by varanthir**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # read-file-lines
-Sync reading lines from file.
-Made for fun and to learn.
-Require: Node with npm
+Reading lines from file. Made for fun and to learn.
+
+It's written in [TypeScript](https://www.typescriptlang.org)
```
Comment (ith) [inline README.md]: Add ## Requirements and list all deps

File: src/read-lines/read-lines.ts [hunk 2]
```diff
@@ -1,61 +1,35 @@
...
-
-  fs.closeSync(file)
-
-  return linesToOmit > 0
-    ? Left<ReadLinesError, string>(ReadLinesError.FileTooShort)
-    : Right<ReadLinesError, string>(result.join('\n'))
+export function readLines(filePath: string, offset: number, limit: number): Promise<string[]> {
+  return new Promise<string[]>((resolve, reject) => {
+    let lines: string[] = []
+    let linesToSkip = offset
```
Comment (ith) [inline src/read-lines/read-lines.ts]: I think this let can be avoided. You can support pagination with simple limit & offset

File: src/read-lines/read-lines.ts [hunk 3]
```diff
@@ -1,61 +1,35 @@
...
+          ? ReadLinesError.FileDoesntExists
+          : ReadLinesError.Unknown)
+      })
+
+    const readline = Readline.createInterface({ input: fileReadStream })
+      .on('line', (line) => {
+        if (linesToSkip > 0) {
+          linesToSkip -= 1
+        } else {
+          lines = lines.concat(line)
```
Comment (ith) [inline src/read-lines/read-lines.ts]: If I remember correctly line is a `string`, not `array` and you're using `concat`. Why not `push`? Yes I know concat is immutable and push isn't. But in the end you're mutating: `let lines`.

Immutable.js?

**Comment by varanthir** [comment]
Please do not review, I found a bug.

File: src/index.ts [hunk 4]
```diff
@@ -1,19 +1,14 @@
...
 const filePath = process.argv[2]
-const offset = argToInt(process.argv[3], 0, 0)
-const limit = argToInt(process.argv[4], 1, 0)
-
-if (!filePath) {
-  console.log('Error: File path not specified')
-  process.exit()
-}
+const offset = process.argv[3]
+const limit = process.argv[4]
```
Comment (kpudlik) [inline src/index.ts]: use destructuring:

`[,, filePath, offset, limit] = process`

File: src/read-lines/utils.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Maybe } from 'monet'
+
+export function minInt(value: string, minVal = 0): Maybe<number> {
+  return Maybe.fromNull(value)
+    .map(val => val.trim())
+    .filter(val => val !== '')
+    .map(Number)
+    .filter(num => !Number.isNaN(num))
```
Comment (kpudlik) [inline src/read-lines/utils.ts]: check if Monet has `.filterNot` then you can `.filterNot(Number.isNaN)`

File: src/read-lines/utils.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { Maybe } from 'monet'
+
+export function minInt(value: string, minVal = 0): Maybe<number> {
+  return Maybe.fromNull(value)
+    .map(val => val.trim())
+    .filter(val => val !== '')
```
Comment (kpudlik) [inline src/read-lines/utils.ts]: Use `Maybe.fromFalsy` and check if line 6 is necesarry then

File: src/index.ts (hunk 4 above)
Comment (varanthir) [reply src/index.ts]: Done

File: src/read-lines/utils.ts (hunk 5 above)
Comment (varanthir) [reply src/read-lines/utils.ts]: Monet hasn't :(

File: src/read-lines/utils.ts (hunk 6 above)
Comment (varanthir) [reply src/read-lines/utils.ts]: `.fromFalsy` + `delete line 6`: function returns `Some(0)` for `value = '        '`

**Pull request merged**